### PR TITLE
Added new functionalities to hint.py

### PR DESCRIPTION
**Pull request by erasedbird**
Added last (last char), count (number of char), and all (return all three pieces of info)

### CODE REVIEW

File: sciolyid/cogs/hint.py [hunk 1]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
         self.bot = bot
 
     # give hint
-    @commands.command(help="- Gives first letter of current image", aliases=["h"])
+    @commands.command(help="- Gives first letter of current image", usage="[count|c last|l all|a]", aliases=["h"])
     @commands.check(CustomCooldown(3.0, bucket=commands.BucketType.channel))
-    async def hint(self, ctx):
-        logger.info("command: hint")
+    async def hint(self, ctx, *args):
+        logger.info("command: hint" + str(len(args)))
```
Comment (tctree333) [inline sciolyid/cogs/hint.py:31]: i don't think you need to log the arg count here

File: sciolyid/cogs/hint.py [hunk 2]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
...
 
         current_item = database.hget(f"channel:{ctx.channel.id}", "item").decode(
             "utf-8"
         )
         if current_item != "":  # check if there is item
-            await ctx.send(f"The first letter is {current_item[0]}")
+            if(len(args) == 0):
+                await ctx.send(f"The first letter is {current_item[0]}.")
+            elif (args[0] == "count" or args[0] == "c"):
+                await ctx.send(f"The answer has {str(len(current_item))} letters.")
```
Comment (tctree333) [inline sciolyid/cogs/hint.py:40]: i don't think you need to use `str()` if you're already using a format string

File: sciolyid/cogs/hint.py [hunk 3]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
...
+            elif (args[0] == "last" or args[0] == "l"):
+                await ctx.send(f"The last letter is {current_item[len(current_item)-1]}.")
```
Comment (tctree333) [inline sciolyid/cogs/hint.py]: you can just index with `[-1]` to get the last letter I think

File: sciolyid/cogs/hint.py [hunk 4]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
...
+            elif (args[0] == "all" or args[0] == "a"):
+                blanks = len(current_item)-2
```
Comment (tctree333) [inline sciolyid/cogs/hint.py]: are you using this variable

File: sciolyid/cogs/hint.py [hunk 5]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
...
+                the_hint = "`"+current_item[0]
+                for i in range(1, len(current_item)-1):
+                    if(current_item[i]!= " "):
+                        the_hint += " _ "
+                    else:
+                        the_hint+="   "
+                await ctx.send(the_hint+ current_item[len(current_item)-1] + "`")
```
Comment (tctree333) [inline sciolyid/cogs/hint.py]: you can just index with `[-1]`

File: sciolyid/cogs/hint.py [hunk 6]
```diff
@@ -25,16 +25,32 @@ def __init__(self, bot):
...
+            if(len(args) == 0):
+                await ctx.send(f"The first letter is {current_item[0]}.")
+            elif (args[0] == "count" or args[0] == "c"):
+                await ctx.send(f"The answer has {str(len(current_item))} letters.")
+            elif (args[0] == "last" or args[0] == "l"):
+                await ctx.send(f"The last letter is {current_item[len(current_item)-1]}.")
+            elif (args[0] == "all" or args[0] == "a"):
+                blanks = len(current_item)-2
+                the_hint = "`"+current_item[0]
+                for i in range(1, len(current_item)-1):
```
Comment (tctree333) [inline sciolyid/cogs/hint.py]: i think it might be cleaner if you just iterate the string? like `for letter in current_item[1:-1]`

**Review by tctree333 — CHANGES_REQUESTED** [review]

File: sciolyid/cogs/hint.py (hunk 2 above)
Comment (erasedbird) [reply sciolyid/cogs/hint.py:40]: It was giving me an error actually

File: sciolyid/cogs/hint.py (hunk 4 above)
Comment (erasedbird) [reply sciolyid/cogs/hint.py]: Not anymore, let me delete

File: sciolyid/cogs/hint.py (hunk 3 above)
Comment (erasedbird) [reply sciolyid/cogs/hint.py]: Got it

File: sciolyid/cogs/hint.py (hunk 5 above)
Comment (erasedbird) [reply sciolyid/cogs/hint.py]: Fixed

File: sciolyid/cogs/hint.py (hunk 6 above)
Comment (erasedbird) [reply sciolyid/cogs/hint.py]: Fixed

**Review by tctree333 — APPROVED** [review]

**Pull request merged**